src/main/php/lang/ast/Tokens.class.php

```diff
@@ -118,7 +118,7 @@ public function iterator($language) {
         yield new Token($language->symbol('(literal)'), 'string', $string, $line);
         $line+= substr_count($string, "\n");
       } else if ('$' === $token) {
-        yield new Token($language->symbol('(variable)'), 'variable', '$'.$next(self::DELIMITERS), $line);
+        yield new Token($language->symbol('(variable)'), 'variable', '$', $line);
```

Comment:
For any parser implementation, it's now necessary to forward once more to get the variable name!